mailpoet/lib/Newsletter/Scheduler/WelcomeScheduler.php

```diff
@@ -118,28 +119,31 @@ public function createWelcomeNotificationSendingTask(NewsletterEntity $newslette
     if (!empty($previouslyScheduledNotification)) {
       return;
     }
-    $sendingTask = SendingTask::create();
-    $sendingTask->newsletterId = $newsletter->getId();
-    $sendingTask->setSubscribers([$subscriberId]);
-    $sendingTask->status = SendingQueueEntity::STATUS_SCHEDULED;
-    $sendingTask->priority = SendingQueueEntity::PRIORITY_HIGH;
-    $sendingTask->scheduledAt = $this->scheduler->getScheduledTimeWithDelay(
+
+    // task
+    $task = new ScheduledTaskEntity();
+    $task->setType(SendingQueue::TASK_TYPE);
+    $task->setStatus(ScheduledTaskEntity::STATUS_SCHEDULED);
+    $task->setPriority(ScheduledTaskEntity::PRIORITY_HIGH);
+    $task->setScheduledAt($this->scheduler->getScheduledTimeWithDelay(
       $newsletter->getOptionValue(NewsletterOptionFieldEntity::NAME_AFTER_TIME_TYPE),
       $newsletter->getOptionValue(NewsletterOptionFieldEntity::NAME_AFTER_TIME_NUMBER)
-    );
-
-    $savedSendingTask = $sendingTask->save();
-
-    // Refreshing this entity here is needed while we are still using Paris to create the scheduled tasks and queues
-    // in the code above using \MailPoet\Tasks\Sending class. Doing this should avoid bugs where the loaded entity contain
-    // stale data after the corresponding entry in the database is updated using Paris. This code can be removed once
-    // https://mailpoet.atlassian.net/browse/MAILPOET-4375 is finished. Currently, if this code is removed a few integration
-    // tests fail (see https://app.circleci.com/pipelines/github/mailpoet/mailpoet/14806/workflows/0d441848-16db-461a-88ec-87bed101fe36/jobs/251385/tests#failed-test-0).
-    $scheduledTaskEntity = $this->scheduledTasksRepository->findOneScheduledByNewsletterAndSubscriber($newsletter, $subscriber);
-    if ($scheduledTaskEntity instanceof ScheduledTaskEntity) {
-      $this->scheduledTasksRepository->refresh($scheduledTaskEntity);
-    }
-
-    return $savedSendingTask;
+    ));
+    $this->entityManager->persist($task);
+
+    // queue
+    $queue = new SendingQueueEntity();
+    $queue->setTask($task);
+    $queue->setNewsletter($newsletter);
+    $queue->setSubscribers((string)$subscriberId);
```

Comment:
Do you happen to know what's this column for? Seems to be a string (comma separated IDs?), but couldn't find any usages of `getSubscribers()` on the queue entity. It's the task entity where we keep the relationships, but I'm not sure what the queue subscribers column is for, and whether I should set it here or not.

Reply:
You're right, I think I added that by mistake instead of setting scheduled task subscribers to the `$task`, then noticed those were missing, and then added those too. This indeed seems to do nothing.

I checked all usages and removed them: https://github.com/mailpoet/mailpoet/pull/5241/commits/388b078725a12f93aa95aa0bb736862926640bf7